From 8e81d20b9400aa04ae7804468b88c6f96ae88003 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 3 Aug 2023 12:37:23 +0800
Subject: [PATCH 06/18] CryptoPkg/openssl: UefiAsm.conf update for openssl 3.0

New naming convention for the configs:  UEFI-${efiarch}-${compiler}.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Cc: Jiewen Yao <jiewen.yao@intel.com>
Cc: Xiaoyu Lu <xiaoyu1.lu@intel.com>
Cc: Guomin Jiang <guomin.jiang@intel.com>
Reviewed-by: Jiewen Yao <jiewen.yao@intel.com>
Acked-by: Ard Biesheuvel <ardb@kernel.org>
Tested-by: Ard Biesheuvel <ardb@kernel.org>
Tested-by: Brian J. Johnson <brian.johnson@hpe.com>
Tested-by: Kenneth Lautner <klautner@microsoft.com>
(cherry picked from commit ac492c3ead964861a0aaf762155a2a923a8d70a0)
---
 CryptoPkg/Library/OpensslLib/UefiAsm.conf | 41 +++++++----------------
 1 file changed, 12 insertions(+), 29 deletions(-)

diff --git a/CryptoPkg/Library/OpensslLib/UefiAsm.conf b/CryptoPkg/Library/OpensslLib/UefiAsm.conf
index c3940b5886..907582f93e 100644
--- a/CryptoPkg/Library/OpensslLib/UefiAsm.conf
+++ b/CryptoPkg/Library/OpensslLib/UefiAsm.conf
@@ -8,41 +8,24 @@
 ##
 
 my %targets = (
-#### UEFI
-    "UEFI-x86" => {
-        perlasm_scheme  => "win32n",
-        # inherit_from    => [ "UEFI",  asm("x86_asm") ],
+    "UEFI-IA32-MSFT" => {
         inherit_from    => [ "UEFI" ],
-        cpuid_asm_src   => "x86cpuid.s",
-        aes_asm_src     => "aes_core.c aes_cbc.c vpaes-x86.s aesni-x86.s",
-        sha1_asm_src    => "sha1-586.s sha256-586.s sha512-586.s",
-        modes_asm_src   => "ghash-x86.s",
+        perlasm_scheme  => "win32n",
+        asm_arch        => "x86",
     },
-    "UEFI-x86-GCC" => {
-        perlasm_scheme  => "elf",
-        # inherit_from    => [ "UEFI",  asm("x86_asm") ],
+    "UEFI-IA32-GCC" => {
         inherit_from    => [ "UEFI" ],
-        cpuid_asm_src   => "x86cpuid.s",
-        aes_asm_src     => "aes_core.c aes_cbc.c vpaes-x86.s aesni-x86.s",
-        sha1_asm_src    => "sha1-586.s sha256-586.s sha512-586.s",
-        modes_asm_src   => "ghash-x86.s",
+        perlasm_scheme  => "elf",
+        asm_arch        => "x86",
     },
-    "UEFI-x86_64" => {
-        perlasm_scheme  => "nasm",
-        # inherit_from    => [ "UEFI",  asm("x86_64_asm") ],
+    "UEFI-X64-MSFT" => {
         inherit_from    => [ "UEFI" ],
-        cpuid_asm_src   => "x86_64cpuid.s",
-        aes_asm_src     => "aes_core.c aes_cbc.c vpaes-x86_64.s aesni-x86_64.s aesni-sha1-x86_64.s aesni-sha256-x86_64.s aesni-mb-x86_64.s",
-        sha1_asm_src    => "sha1-x86_64.s sha256-x86_64.s sha512-x86_64.s sha1-mb-x86_64.s sha256-mb-x86_64.s",
-        modes_asm_src   => "ghash-x86_64.s aesni-gcm-x86_64.s",
+        perlasm_scheme  => "nasm",
+        asm_arch        => "x86_64",
     },
-    "UEFI-x86_64-GCC" => {
-        perlasm_scheme  => "elf",
-        # inherit_from    => [ "UEFI",  asm("x86_64_asm") ],
+    "UEFI-X64-GCC" => {
         inherit_from    => [ "UEFI" ],
-        cpuid_asm_src   => "x86_64cpuid.s",
-        aes_asm_src     => "aes_core.c aes_cbc.c vpaes-x86_64.s aesni-x86_64.s aesni-sha1-x86_64.s aesni-sha256-x86_64.s aesni-mb-x86_64.s",
-        sha1_asm_src    => "sha1-x86_64.s sha256-x86_64.s sha512-x86_64.s sha1-mb-x86_64.s sha256-mb-x86_64.s",
-        modes_asm_src   => "ghash-x86_64.s aesni-gcm-x86_64.s",
+        perlasm_scheme  => "elf",
+        asm_arch        => "x86_64",
     },
 );
-- 
2.43.0

